src/map.cpp

```diff
@@ -6038,6 +6038,44 @@ bool map::sees( const tripoint &F, const tripoint &T, const int range, int &bres
     return visible;
 }
 
+int map::obstacle_coverage( const tripoint &loc1, const tripoint &loc2 )
+{
+    if( move_cost( loc2 ) > 2 ) { // Can't hide if you are standing on something.
```

Comment:
This check won't always make sense. For example, I think you still can safely hide even while standing on a sand with `move_cost` 3.

Reply:
Sand is the case where "slows down" is not equal to "has an obstacle", so I don't know how to resolve that. It's too specific, and sand is not that frequent in the game, especially furniture near sand tiles. Since I lack skills and that issue doesn't seem critical, can we just call it a day?

Reply:
If sand is the only type of such terrain, then we could just close eyes on that. But I think you should check if there are other such terrains, just to be sure.

Reply:
Okay, I'll edit this comment when I'm done.